arithmetization/src/main/java/net/consensys/linea/zktracer/module/bin/BinOperation.java

```diff
@@ -34,37 +34,24 @@
 
 @Getter
 @Accessors(fluent = true)
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = false)
 public class BinOperation extends ModuleOperation {
-  public BinOperation(OpCode opCode, BaseBytes arg1, BaseBytes arg2) {
-    this.opCode = opCode;
-    this.arg1 = arg1;
-    this.arg2 = arg2;
-  }
-
   private static final int LIMB_SIZE = 16;
-  private final OpCode opCode;
-  private final BaseBytes arg1;
-  private final BaseBytes arg2;
+
+  @EqualsAndHashCode.Include private final OpCode opCode;
+  @EqualsAndHashCode.Include private final BaseBytes arg1;
+  @EqualsAndHashCode.Include private final BaseBytes arg2;
   private List<Boolean> lastEightBits = List.of(false);
   private boolean bit4 = false;
   private int low4 = 0;
   private boolean isSmall = false;
   private int pivotThreshold = 0;
   private int pivot = 0;
 
-  @Override
-  public int hashCode() {
-    return Objects.hashCode(this.opCode, this.arg1, this.arg2);
-  }
-
-  @Override
-  public boolean equals(Object o) {
-    if (this == o) return true;
-    if (o == null || getClass() != o.getClass()) return false;
-    final BinOperation that = (BinOperation) o;
-    return java.util.Objects.equals(opCode, that.opCode)
-        && java.util.Objects.equals(arg1, that.arg1)
-        && java.util.Objects.equals(arg2, that.arg2);
+  public BinOperation(OpCode opCode, BaseBytes arg1, BaseBytes arg2) {
```

Comment:
Could that be a `@RequiredArgsConstructor`?

Reply:
Yes, I was just worried not to break something, if I do it, but actually there is no issue, so I will fix it.